### PR TITLE
Ports some hud updates from TG

### PR DESCRIPTION
**Pull request by Aurorablade**
ports https://github.com/tgstation/tgstation/pull/32967
ports also (I LOST MY LINKS AND I AM LAZY) all living mods but silicons and ghosts have a med hud, Swarmers are on diag huds.

https://puu.sh/yvyW7/2fe7a7e119.gif

:cl: Fethas
rscadd: TG Port, ais now have an advanced diagnostic hud that lets them see the path a bot is taking.
tweak: Bots get access for 60 seconds to all doors to get to thier locations.
rscadd:Pai controlled bots will get a notification they are being called.
rscadd:all living mobs now can have a med hud.
/:cl:

### CODE REVIEW

**Comment by Aurorablade** [comment]
HOLD ON BOYS WE GOING DEEPER!

**Comment by taukausanake** [comment]
> all living mobs now can have a med hud

Can you explain this in more detail and/or find the PR you're porting it from? I don't think I understand it enough

**Comment by Aurorablade** [comment]
right now only living CARBON mobs can have a med hud. this makes it so all LIVING mobs can have a medhud.

**Comment by taukausanake** [comment]
I'm reading that as either living mobs can have a medhud view or medhud overlay applied to them. Otherwise I am still a little confused. I guess this means that simple mobs could have the overlay rather than just player mobs

**Comment by Aurorablade** [comment]
correct. simple mobs are a living mob but they are not a CARBON mob.

**Comment by taukausanake** [comment]
Okay. Not sure why ghosts and silicons (I'm thinking borgs and IPCs in that regard) would be denied the overlay as I don't think they were applied before anyways? At least for borgs. Would this mean IPCs need a diaghud to see health?
Is the ghost check because of cult ghosts?

**Comment by Aurorablade** [comment]
ipcs are not a silicon they are a carbon. and ghosts..are not alive..why would they need a med hud applied to them.?

**Comment by taukausanake** [comment]
The way things are worded confuses me. Obviously ghosts wouldn't have HUD overlays to start with. The reason you're pointing them out is the confusing part. The same for borgs that don't use a medhud; they use the diaghud. I guess the only distinction for that though is that borgs fall under living/silicon and saying all living counts them.
Now I'm wondering why IPCs are carbon based

**Comment by Fox-McCloud** [comment]
This just means that when you're wearing a medical HUD you can see the health of simple animals, as well, basically.

**Comment by taukausanake** [comment]
@Fox-McCloud That is SO much easier to understand, thanks

**Comment by alexkar598** [comment]
Finally i can see the health of the poor slimes I kill

**Comment by adamkad1** [comment]
medhud for simple mobs, cool! just another feature i make frequent use of in tg but solely for purposes of lavalanding

**Comment by alexkar598** [comment]
Can you keep the old medihud health bars?

It's important to know if a patient is almost dead or just in soft critical

Right now with these new huds you can only tell if a patient(in crit) has more or less than -50%

**Comment by alexkar598** [comment]
Thank you

File: code/game/data_huds.dm [hunk 1]
```diff
@@ -76,51 +76,54 @@
...
+			return "health-60"
+		if(-80 to -70)
+			return "health-70" //Doc?
+		if(-90 to -80)
+			return "health-80" //Hey, doc?
+		if(-100 to -90)
+			return "health-90" //HURRY UP, DOC!
 		else
-			return "health-100"
+			return "health-100" //doc u had 1 job
```
Comment (taukausanake) [inline code/game/data_huds.dm]: :rofl:

**Comment by KingPhilipIII** [comment]
When will we port the vastly superior TG ghost hud that lets them see all three at once.

File: code/game/gamemodes/miniantags/bot_swarm/swarmer.dm [hunk 2]
```diff
@@ -127,8 +128,22 @@
...
 	verbs -= /mob/living/verb/pulled
+	for(var/datum/atom_hud/data/diagnostic/diag_hud in huds)
+		diag_hud.add_to_hud(src)
 	updatename()
 
+/mob/living/simple_animal/hostile/swarmer/med_hud_set_health()
+	var/image/holder = hud_list[DIAG_HUD]
+	var/icon/I = icon(icon, icon_state, dir)
+	holder.pixel_y = I.Height() - world.icon_size
+	holder.icon_state = "huddiag[RoundDiagBar(health/maxHealth)]"
```
Comment (tigercat2000) [inline code/game/gamemodes/miniantags/bot_swarm/swarmer.dm]: Spacing around math operators.

File: code/modules/mob/living/simple_animal/bot/bot.dm [hunk 3]
```diff
@@ -75,14 +75,21 @@
...
 	"Waiting for clear path", "Calculating navigation path", "Pinging beacon network", "Unable to reach destination")
 
-	hud_possible = list(DIAG_STAT_HUD, DIAG_BOT_HUD, DIAG_HUD) //Diagnostic HUD views
+	var/datum/atom_hud/data/bot_path/path_hud = new /datum/atom_hud/data/bot_path()
+	var/path_image_icon = 'icons/obj/aibots.dmi'
+	var/path_image_icon_state = "path_indicator"
+	var/path_image_color = "#FFFFFF"
+	var/reset_access_timer_id
+
+	hud_possible = list(DIAG_STAT_HUD, DIAG_BOT_HUD, DIAG_HUD, DIAG_PATH_HUD = HUD_LIST_LIST)//Diagnostic HUD views
```
Comment (tigercat2000) [inline code/modules/mob/living/simple_animal/bot/bot.dm]: What does `DIAG_PATH_HUD = HUD_LIST_LIST` do...?

Comment (Aurorablade) [reply code/modules/mob/living/simple_animal/bot/bot.dm]: i am looking at this right i think it has to do with the path images being created
//by default everything in the hud_list of an atom is an image
//a value in hud_list with one of these will change that behavior
 #define HUD_LIST_LIST 1

**Pull request merged**